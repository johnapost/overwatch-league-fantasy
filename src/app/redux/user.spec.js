import reducer, { defaultState, userLogin, userLogout } from "./user";

describe("user", () => {
  describe("action creators", () => {
    it("userLogin should return action", () => {
      expect(userLogin("123")).toMatchSnapshot();
    });

    it("userLogout should return action", () => {
      expect(userLogout()).toMatchSnapshot();
    });
  });

  describe("reducer", () => {
    it("should return default state", () => {
      expect(reducer(undefined, { type: "" })).toMatchSnapshot();
    });

    it("should update user when logging in and out", () => {
      const logIn = reducer(defaultState, userLogin("123"));
      expect(logIn).toMatchSnapshot();

      const setProfile = reducer(logIn, {
        type: "@@reactReduxFirebase/SET_PROFILE",
        profile: { displayName: "testUser" }
      });
      expect(setProfile).toMatchSnapshot();

      const logOut = reducer(setProfile, userLogout());
      expect(logOut).toMatchSnapshot();
    });
  });
});
