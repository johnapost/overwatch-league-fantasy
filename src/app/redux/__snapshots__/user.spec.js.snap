// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user action creators userLogin should return action 1`] = `
Object {
  "type": "LOGIN",
  "uid": "123",
}
`;

exports[`user action creators userLogout should return action 1`] = `
Object {
  "type": "LOGOUT",
}
`;

exports[`user reducer should return default state 1`] = `
Object {
  "displayName": null,
  "uid": null,
}
`;

exports[`user reducer should update user when logging in 1`] = `
Object {
  "displayName": null,
  "uid": "123",
}
`;

exports[`user reducer should update user when logging in 2`] = `
Object {
  "displayName": "testUser",
  "uid": "123",
}
`;

exports[`user reducer should update user when logging in and out 1`] = `
Object {
  "displayName": null,
  "uid": "123",
}
`;

exports[`user reducer should update user when logging in and out 2`] = `
Object {
  "displayName": "testUser",
  "uid": "123",
}
`;

exports[`user reducer should update user when logging in and out 3`] = `
Object {
  "displayName": null,
  "uid": null,
}
`;
